### PR TITLE
enhance getStateAs for DimmerItem and RollershutterItem

### PR DESCRIPTION
**Pull request by maggu2810**
This commit also contains work of 'Chris Jackson <chris@cd-jackson.com>'
who started the SwitchItemTest and DimmerItemTest implementation.

Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=467739
Signed-off-by: Markus Rathgeb <maggu2810@gmail.com>

### CODE REVIEW

**Comment by maggu2810** [comment]
This should continue (replace) the work of #311

File: bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    private static BigDecimal getState(final Item item, Class<? extends State> typeClass) {
+        final State state = item.getStateAs(typeClass);
+        final String str = state.toString();
+        final BigDecimal result = new BigDecimal(str);
+        return result;
+    }
+
+    @Test
+    public void GetAsPercentFromPercent() {
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java]: Did you name the tests on with a capital letter on purpose? Looks rather unusual to me...

Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java]: Copy Paste from the one test of @cdjackson

Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java]: That's no excuse :-P

Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java]: You are right, it's even worse. Sorry ;-)

**Comment by maggu2810** [comment]
I fixed the case sensitivity.

File: bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/SwitchItemTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.junit.Test;
+
+/**
+ * @author Chris Jackson - Initial contribution
+ */
+public class SwitchItemTest {
+
+    @Test
+    public void GetAsPercentFromSwitch() {
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/items/SwitchItemTest.java]: All, but one ;-)

**Comment by maggu2810** [comment]
I find this really annoying.

**Comment by kaikreuzer** [comment]
Me or this? Thanks anyhow :-)

**Pull request merged**